src/utils/material-utils.js

```diff
@@ -17,3 +17,144 @@ export function mapMaterials(object3D, fn) {
     return fn(object3D.material);
   }
 }
+
+export function convertStandardMaterial(source, quality) {
+  if (!source.isMeshStandardMaterial) {
+    return source;
+  }
+
+  if (quality === "medium") {
+    const material = new THREE.MeshPhongMaterial();
+
+    THREE.Material.prototype.copy.call(material, source);
+
+    material.color.copy(source.color);
+
+    material.map = source.map;
+
+    material.lightMap = source.lightMap;
+    material.lightMapIntensity = source.lightMapIntensity;
+
+    material.aoMap = source.aoMap;
+    material.aoMapIntensity = source.aoMapIntensity;
+
+    material.emissive.copy(source.emissive);
+    material.emissiveMap = source.emissiveMap;
+    material.emissiveIntensity = source.emissiveIntensity;
+
+    material.normalMapType = source.normalMapType;
+    material.normalMap = source.normalMap;
+    material.normalScale.copy(source.normalScale);
+
+    material.onBeforeCompile = shader => {
```

Comment:
A quick comment explaining why we are patching the shader would be helpful here I think.